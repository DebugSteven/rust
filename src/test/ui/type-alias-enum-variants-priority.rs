#![feature(type_alias_enum_variants)]

enum E {
    V(u8)
}

trait Tr {
    type V;
    fn f() -> Self::V;
}

impl Tr for E {
    type V = u8;
    fn f() -> Self::V { loop {} } //~ ERROR type vs variant ambiguity
}

fn main() {}
