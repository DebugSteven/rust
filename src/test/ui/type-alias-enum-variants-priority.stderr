error: type vs variant ambiguity
  --> $DIR/type-alias-enum-variants-priority.rs:14:15
   |
LL |     fn f() -> Self::V { loop {} } //~ ERROR type vs variant ambiguity
   |               ^^^^^^^
   |
   = note: #[forbid(type_vs_variant_ambiguity)] on by default

error: aborting due to previous error

